src/main/java/org/commcare/formplayer/services/CaseSearchHelper.java

```diff
@@ -39,16 +38,16 @@
 
     public ExternalDataInstance getSearchDataInstance(FormplayerQueryScreen screen,
                                                       boolean skipDefaultPromptValues) {
-        URI uri = screen.getUri(skipDefaultPromptValues);
+        MultiValueMap<String, String> queryParams = screen.getRequestData(skipDefaultPromptValues);
         Cache cache = cacheManager.getCache("case_search");
-        String cacheKey = getCacheKey(uri);
+        String cacheKey = getCacheKey(queryParams.hashCode());
```

Comment:
this hash code could in theory have collisions - is that something we need to worry about?  Any other ideas for getting a unique code to use to identify a `queryParams` state?

Reply:
I think we would also want to include the `uri` here as part of cache key so that 2 different urls with same params don't get treated as equals.

Reply:
we can probably append all key-value pairs as it is in the `cacheKey` instead of the hash ? Collisions are very unlikely to happen though a bug like that would be a hard one to identify in future.

Reply:
Good point - I guess the domain is only provided in the URI, not in the parameters, and that'd be important to vary on.

I can append the key-value pairs directly - do you think there's any risk of the cache key being too big and overflowing something?  In particular, the point of switching this to a post request is to support parameter sets that are many thousands of characters long

Reply:
the cache key is `String`. So I think it's ok as long as we don't overshoot the `String` length limits